client/landing/stepper/declarative-flow/site-migration-flow.ts

```diff
@@ -15,22 +16,27 @@ import { HOW_TO_MIGRATE_OPTIONS } from '../constants';
 import { useIsSiteAdmin } from '../hooks/use-is-site-admin';
 import { useSiteData } from '../hooks/use-site-data';
 import { useSiteSlugParam } from '../hooks/use-site-slug-param';
-import { USER_STORE, ONBOARD_STORE, SITE_STORE } from '../stores';
+import { USER_STORE, SITE_STORE, ONBOARD_STORE } from '../stores';
 import { goToCheckout } from '../utils/checkout';
-import { recordSubmitStep } from './internals/analytics/record-submit-step';
 import { STEPS } from './internals/steps';
 import { getSiteIdParam } from './internals/steps-repository/import/util';
 import { type SiteMigrationIdentifyAction } from './internals/steps-repository/site-migration-identify';
 import { AssertConditionState } from './internals/types';
 import type { AssertConditionResult, Flow, ProvidedDependencies } from './internals/types';
-import type { OnboardSelect, SiteSelect, UserSelect } from '@automattic/data-stores';
 
 const FLOW_NAME = 'site-migration';
 
 const siteMigration: Flow = {
 	name: FLOW_NAME,
 	isSignupFlow: false,
 
+	useSideEffect() {
+		const { setIntent } = useDispatch( ONBOARD_STORE );
+		useEffect( () => {
+			setIntent( Onboard.SiteIntent.SiteMigration );
+		}, [] );
+	},
```

Comment:
@Automattic/dotcom-stepper I'm raising same concern here with https://github.com/Automattic/wp-calypso/pull/93178 and https://github.com/Automattic/wp-calypso/pull/93178#discussion_r1699819153

Here are a few questions to get some answer out of:

1. What is the intention with `isSignupFlow: false`? Do these intend **not** to create new sites in the end?
2. What is the intention with `Onboard.SiteIntent` - is it meant to map directly to the recorded intent in the site's meta e.g. what you'd get back from [useSiteIntent](https://github.com/Automattic/wp-calypso/blob/trunk/packages/data-stores/src/queries/use-site-intent.ts)? If not, then perhaps a different name might work to avoid confusion?
3. Is it ok design-wise to set intent from here and just reset it from child/consuming flows (like `hosted-site-migration`)?

Reply:
I can answer one of these:

> What is the intention with isSignupFlow: false? Do these intend not to create new sites in the end?

IIRC, this is meant to not fire `calypso_signup_start` since this flow is meant to configure a site, rather than create one.

Reply:
Thanks for the reply. Same discussion in other PR https://github.com/Automattic/wp-calypso/pull/93178#discussion_r1699819153 from which ended up creating https://github.com/Automattic/wp-calypso/issues/93242 

as mentioned there:

> It's something to follow up on. For now, I intend to keep whatever was recorded to be recorded as previously. The risk may be adding more values to SiteIntent that may never get persisted as site intent. Not sure how else to have a concrete type for the Tracks recorders without over-engineering.